x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/LookupJoin.java

```diff
@@ -71,4 +76,40 @@ protected NodeInfo<Join> info() {
             config().rightFields()
         );
     }
+
+    @Override
+    public void postAnalysisVerification(Failures failures) {
+        super.postAnalysisVerification(failures);
+        right().forEachDown(EsRelation.class, esr -> {
+
+            if (esr.indexMode().equals(IndexMode.LOOKUP)) {
+                var indexNameWithModes = esr.indexNameWithModes();
+                if (indexNameWithModes.size() != 1) {
+                    failures.add(
+                        fail(
+                            esr,
+                            "invalid [{}] resolution in lookup mode to an index in [{}] indices",
+                            esr.indexPattern(),
+                            indexNameWithModes.size()
+                        )
+                    );
+                } else if (indexNameWithModes.values().iterator().next() != IndexMode.LOOKUP) {
+                    failures.add(
+                        fail(
+                            esr,
+                            "invalid [{}] resolution in lookup mode to an index in [{}] mode",
+                            esr.indexPattern(),
+                            indexNameWithModes.values().iterator().next()
+                        )
+                    );
+                }
+
+            }
+
+            // this check is crucial for security: ES|QL would use the concrete indices, so it would bypass the security on the alias
```

Comment:
Note: After debugging a little, I believe we wouldn't actually bypass security, but we would "forget" about the alias and instead act as if the concrete index was requested - which the user may or may not have permissions on. But it's still better to leave this comment as-is in to emphasize that changing this check should be scary.